packages/components/src/query-controls/index.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import AuthorSelect from './author-select';
+import CategorySelect from './category-select';
+import FormTokenField from '../form-token-field';
+import RangeControl from '../range-control';
+import SelectControl from '../select-control';
+import type { QueryControlsProps } from './types';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 100;
+const MAX_CATEGORIES_SUGGESTIONS = 20;
+
+/**
+ * Controls to query for posts.
+ *
+ * ```jsx
+ * const MyQueryControls = () => (
+ *   <QueryControls
+ *     { ...{ maxItems, minItems, numberOfItems, order, orderBy } }
+ *     onOrderByChange={ ( newOrderBy ) => {
+ *       updateQuery( { orderBy: newOrderBy } )
+ *     }
+ *     onOrderChange={ ( newOrder ) => {
+ *       updateQuery( { order: newOrder } )
+ *     }
+ *     categoriesList={ categories }
+ *     selectedCategoryId={ category }
+ *     onCategoryChange={ ( newCategory ) => {
+ *       updateQuery( { category: newCategory } )
+ *     }
+ *     onNumberOfItemsChange={ ( newNumberOfItems ) => {
+ *       updateQuery( { numberOfItems: newNumberOfItems } )
+ *     } }
+ *   />
+ * );
+ * ```
+ */
+export function QueryControls( {
+	authorList,
+	selectedAuthorId,
+	categoriesList,
+	selectedCategoryId,
+	categorySuggestions,
+	selectedCategories,
+	numberOfItems,
+	order,
+	orderBy,
+	maxItems = DEFAULT_MAX_ITEMS,
+	minItems = DEFAULT_MIN_ITEMS,
+	onCategoryChange,
+	onAuthorChange,
+	onNumberOfItemsChange,
+	onOrderChange,
+	onOrderByChange,
+}: QueryControlsProps ) {
+	return (
+		<>
```

Comment:
I added this wrapping `<>` to prevent this TS error in `stories/index.tsx`:

>(Element | undefined)[]' does not satisfy the constraint 'keyof IntrinsicElements | JSXElementConstructor<any>'.ts(2344)

<img width="1187" alt="Screen Shot 2022-12-24 at 12 53 03 AM" src="https://user-images.githubusercontent.com/4063887/209425123-b8c93f64-e1c4-41d7-89c8-bcb737f3639e.png">